tests/paramgrill.c

```diff
@@ -632,131 +1391,1112 @@ int benchFiles(const char** fileNamesTable, int nbFiles)
+                                    winner.params = CParams;
+                                }
+                            }
+                        }
+                    }
+                } //best, -1, +1, ..., 
+
+            } else if (paramTarget.strategy == 0) { //constrained
+                int feas = 0, i, j; 
+                for(j = 1; j < 10; j++) {
+                    for(i = 1; i <= maxSeeds; i++) {
+                        int varLenNew = sanitizeVarArray(varLen, varArray, varNew, i);
+                        ZSTD_compressionParameters candidateParams = paramTarget;
+                        BMK_result_t candidate;
+                        int ec;
+                        randomConstrainedParams(&candidateParams, varNew, varLenNew, allMT[i]);
+                        cParamZeroMin(&candidateParams);
+                        candidateParams = sanitizeParams(candidateParams);
+                        ec = BMK_benchParam(&candidate, origBuff, benchedSize, fileSizes, (U32)nbFiles, ctx, dctx, candidateParams);
+                        
+                        if(!ec) {
+                            if(feas) {
+                                if(feasible(candidate, relaxTarget(target)) && objective_lt(winner.result, candidate)) {
+                                    winner.result = candidate;
+                                    winner.params = candidateParams;
+                                    BMK_printWinner(stdout, i, winner.result, winner.params, benchedSize);
+                                }
+                            } else {
+                                if(feasible(candidate, relaxTarget(target))) {
+                                    feas = 1;
+                                    winner.result = candidate;
+                                    winner.params = candidateParams;
+                                    BMK_printWinner(stdout, i, winner.result, winner.params, benchedSize);
+
+                                } else {
+                                    if(resultScore(candidate, benchedSize, target) > resultScore(winner.result, benchedSize, target)) {
+                                        winner.result = candidate;
+                                        winner.params = candidateParams;
+                                        BMK_printWinner(stdout, i, winner.result, winner.params, benchedSize);
+                                    }
+                                }
+                            }
+                        }
+
+                    }
                 }
-                if ( (candidate.cSize < winner.result.cSize)
-                   | ((candidate.cSize == winner.result.cSize) & (candidate.cSpeed > winner.result.cSpeed)) )
-                {
-                    winner.params = CParams;
-                    winner.result = candidate;
-                    BMK_printWinner(stdout, i, winner.result, winner.params, benchedSize);
-            }   }
+            }
```

Comment:
when there are so many levels nested,
it becomes hard to understand what is being closed.

Consider indicating in comment to which begin each end refers to,
at least for the outer most ones.